### PR TITLE
chore: Update data format for platform API feature management

### PR DESCRIPTION
**Pull request by smartdev58**
## Description

Updated json data format for Platform API for account creation update API endpoints features flags to accept false values on each feature. #5717 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

https://www.loom.com/share/b0d9452d556f41738924239cba080e92

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb [hunk 1]
```diff
@@ -42,29 +42,24 @@
         json_response = JSON.parse(response.body)
         expect(json_response['name']).to eq('Test Account')
         expect(json_response['locale']).to eq('es')
-        expect(json_response['enabled_features']).to eq(
```
Comment (sojan-official) [inline spec/controllers/platform/api/v1/accounts_controller_spec.rb:45]: I believe this line is intentional and shouldn't be removed. We want to make sure that the default features enabled by installation config should work when an account is created with out any specification through `features` params

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb [hunk 2]
```diff
@@ -144,14 +140,19 @@
 
         patch "/platform/api/v1/accounts/#{account.id}", params: {
           name: 'Test Account',
-          enabled_features: %w[feature_ip_lookup feature_help_center],
+          features: {
```
Comment (sojan-official) [inline spec/controllers/platform/api/v1/accounts_controller_spec.rb:143]: @smartdev58 when specifying  { `feature_name`:  false } , will it disable the feature if it was already enabled ? 
If so, can you update this test case to ensure that?

**Review by sojan-official — CHANGES_REQUESTED** [review]
The new format looks good to me and seems more versatile. 
Added a comment on the patch request. lets process to merge once that is resolved.

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb [hunk 3]
```diff
@@ -42,29 +42,25 @@
...
-        expect(json_response['enabled_features']).to eq(
-          {
-            'agent_management' => true
-          }
-        )
+        expect(json_response['enabled_features']['agent_management']).to be(true)
       end
 
       it 'creates an account with feature flags' do
         InstallationConfig.where(name: 'ACCOUNT_LEVEL_FEATURE_DEFAULTS').first_or_create!(value: [{ 'name' => 'inbox_management',
```
Comment (sojan-official) [inline spec/controllers/platform/api/v1/accounts_controller_spec.rb:49]:
```suggestion
        InstallationConfig.where(name: 'ACCOUNT_LEVEL_FEATURE_DEFAULTS').first_or_create!(value: [{ 'name' => 'disable_branding',
```
let's have this added to ensure that `disable_branding`: false in the params is working properly.

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb (hunk 2 above)
Comment (smartdev58) [reply spec/controllers/platform/api/v1/accounts_controller_spec.rb:143]: > @smartdev58 when specifying { `feature_name`: false } , will it disable the feature if it was already enabled ? If so, can you update this test case to ensure that?

@sojan-official I have already done the update you mentioned for the patch. line: 146: `channel_facebook: false`

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb [hunk 4]
```diff
@@ -144,14 +140,19 @@
 
         patch "/platform/api/v1/accounts/#{account.id}", params: {
           name: 'Test Account',
-          enabled_features: %w[feature_ip_lookup feature_help_center],
+          features: {
+            ip_lookup: true,
+            help_center: true,
+            channel_facebook: false
```
Comment (smartdev58) [inline spec/controllers/platform/api/v1/accounts_controller_spec.rb:146]: @sojan-official Here is what I did.

File: spec/controllers/platform/api/v1/accounts_controller_spec.rb (hunk 3 above)
Comment (smartdev58) [reply spec/controllers/platform/api/v1/accounts_controller_spec.rb:49]: I am done with this suggested changes.

**Review by smartdev58 — COMMENTED** [review]
I am done with the requested changes.

**Comment by smartdev58** [comment]
> The new format looks good to me and seems more versatile. Added a comment on the patch request. lets process to merge once that is resolved.

I am done with the changes in patch request test spec.

**Comment by sojan-official** [comment]
@smartdev58, thanks a lot for the changes.  I have slightly refactored the logic so that the behaviour is consistent for creation and update. With this change, the currently enabled features in an account or features enabled through installation configuration are not affected unless the feature key is explicitly mentioned in features params. 

```
Example: 
Account 1 -> already has inbox_management / channel_facebook enabled
Patch /accounts/1  { features: {  inbox_management: false, channel_email: true } 

Based on the initial PR -> after the patch account will only have: channel_email to be true; the previously enabled channel_facebook is also used to get overwritten.

With the current changes: the account will have both channel_email and channel_facebook enabled after the patch.

```

**Pull request merged**